### PR TITLE
Advanced 065 Primitive: add tagged, quiz, and examples

### PR DESCRIPTION
**Pull request by leogermond**
- adv 065: add quiz multiple primitives
- adv 065: format
- adv 065: add quiz
- adv 065: quiz override
- adv 065: fix quiz include
- adv 065: WIP type freeze gnatDG
- adv 065 primitive: fix quiz
- adv 065: fix example
- adv 065: downcast typo fix
- adv 065: fix quiz

### CODE REVIEW

File: courses/fundamentals_of_ada/adv_065_primitives.rst [hunk 1]
```diff
@@ -27,25 +27,13 @@ Advanced Primitives
...
-      .. code:: Ada
-
-         package P is
-            type T1 is range 1 .. 10;
-            type T2 is (A, B, C);
-
-            procedure Proc (V1 : T1; V2 : T2);
-            function "+" (V1 : T1; V2 : T2) return T1;
-         end P;
+-------------
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_065_primitives.rst:34]: This (Inheritance) is a slide without content

File: courses/fundamentals_of_ada/adv_065_primitives.rst [hunk 2]
```diff
@@ -83,11 +91,153 @@ Implicit Primitive Operations
...
       end P;
       ...
       procedure Main is
          V1, V2 : T1;
       begin
          V1 := V1 + V2;
       end Main;
+
+---------------------------
+Recap. on type derivation
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_065_primitives.rst:105]: I don't think "Recap" is an abbreviation (so no ".") but maybe "Review" would be better understood by non-English speakers

File: courses/fundamentals_of_ada/adv_065_primitives.rst [hunk 3]
```diff
@@ -83,11 +91,153 @@ Implicit Primitive Operations
...
+    + Freeze point rules don't change
+    + Primitives are inherited by child types
+    + Conversion from child to parent possible
+    + Pre-defined set of primitives
+
+        - ``"+", "-" ...`` for numeric types
+        - Comparison operators
+        - Equality except if :ada:`limited`
+
+* Derived type that are **not** :ada:`tagged`
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_065_primitives.rst:119]: "types that are" or "type that is"

File: courses/fundamentals_of_ada/adv_065_primitives.rst [hunk 4]
```diff
@@ -83,11 +91,153 @@ Implicit Primitive Operations
...
+.. code:: Ada
+
+    type Root1 is tagged null record;
+    type Root2 is tagged null record;
+
+    procedure P1_Correct (V1 : Root1; V2 : Root1);
+    procedure P2_Incorrect (V1 : Root1; V2 : Root2); -- FAIL
+
+------------------------------
+Recap. on tagged inheritance
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_065_primitives.rst:214]: "Review"

File: courses/fundamentals_of_ada/quiz/primitives_and_classwide/quiz.rst [hunk 5]
```diff
@@ -0,0 +1,21 @@
+.. code:: Ada
+
+    type T1 is tagged null record;
+    type T2 is new T1 with null record;
+    -- Declarations
+    V : T2;
+
+Which of the following piece(s) of code allow for calling :ada:`Proc (V)`?
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/quiz/primitives_and_classwide/quiz.rst:8]: Should we note that if these procedures are in a different scope, then they are fine?

**Review by frank-at-adacore — CHANGES_REQUESTED** [review]

File: courses/fundamentals_of_ada/adv_065_primitives.rst (hunk 1 above)
Comment (leogermond) [reply courses/fundamentals_of_ada/adv_065_primitives.rst:34]: Indeed, I intended to do a recap of primitive, inheritance, type-class, but this is probably too much, removed it

File: courses/fundamentals_of_ada/adv_065_primitives.rst (hunk 2 above)
Comment (leogermond) [reply courses/fundamentals_of_ada/adv_065_primitives.rst:111]: Got it, interesting, recap does indeed exist, unlike French were we only have the longer-form of recapitulate

File: courses/fundamentals_of_ada/quiz/primitives_and_classwide/quiz.rst (hunk 5 above)
Comment (leogermond) [reply courses/fundamentals_of_ada/quiz/primitives_and_classwide/quiz.rst:8]: For answer C, right?

Comment (frank-at-adacore) [reply courses/fundamentals_of_ada/quiz/primitives_and_classwide/quiz.rst:8]: Yes, but maybe better is a reminder that "primitive" only applies to subprograms in the same scope?

**Pull request merged**